### PR TITLE
Fix shell command "test " cannot differentiate exception and "false"

### LINKED ISSUE #10385: shell command "test " cannot differentiate exception and "false" result
sample command: "test -f some_existing_file"
Problem:
in case of exception, this command returns code=1, implying this is NOT a file, which is not the fact.

[TestCommand.java](https://github.com/Alluxio/alluxio/blob/9dea2a67cad376d32b2d2409e08e9280770b27b3/shell/src/main/java/alluxio/cli/fs/command/TestCommand.java#L165)

proposal:
in case of exception, can we output the exception together with return code=1?
so that we could retry if "exception" is detected in command line output.

### PR DESCRIPTION
**Pull request by codings-dan**
Fix #10385

### CODE REVIEW

File: shell/src/main/java/alluxio/cli/fs/command/TestCommand.java [hunk 1]
```diff
@@ -162,8 +163,12 @@ public int run(CommandLine cl) throws AlluxioException, IOException {
         return -1;
       }
       return testResult ? 0 : 1;
-    } catch (AlluxioException | IOException e) {
+    } catch (FileNotFoundException e) {
       return 1;
+    } catch (IOException e){
```
Comment (gpang) [inline shell/src/main/java/alluxio/cli/fs/command/TestCommand.java:168]:
```suggestion
    } catch (IOException e) {
```

File: shell/src/main/java/alluxio/cli/fs/command/TestCommand.java [hunk 2]
```diff
@@ -162,8 +163,12 @@ public int run(CommandLine cl) throws AlluxioException, IOException {
...
+      return 2;
+    }catch (AlluxioException e){
```
Comment (gpang) [inline shell/src/main/java/alluxio/cli/fs/command/TestCommand.java:170]:
```suggestion
    } catch (AlluxioException e) {
```

**Review by gpang — CHANGES_REQUESTED** [review]
@codings-dan Thanks for the improvement!

**Comment by codings-dan** [comment]
> Automated checks report:
> 
> * Commits associated with Github account: PASS
> * PR title follows the conventions: PASS
> 
> All checks passed!

**Comment by codings-dan** [comment]
@gpang Thanks for reviewing my code

**Comment by codings-dan** [comment]
> > Automated checks report:
> > 
> > * Commits associated with Github account: PASS
> > * PR title follows the conventions: PASS
> > 
> > All checks passed!

**Review by apc999 — APPROVED** [review]

**Comment by LuQQiu** [comment]
```
Error: 1.293 [ERROR] Failures: 
Error: 1.293 [ERROR] alluxio.client.cli.fs.command.TestCommandIntegrationTest.testDirNotEmptyWhenPathNotExist
Error: 1.293 [ERROR]   Run 1: TestCommandIntegrationTest.testDirNotEmptyWhenPathNotExist:92 expected:<1> but was:<3>
Error: 1.293 [ERROR]   Run 2: TestCommandIntegrationTest.testDirNotEmptyWhenPathNotExist:92 expected:<1> but was:<3>
20:24:31.293 [INFO] 
Error: 1.293 [ERROR] alluxio.client.cli.fs.command.TestCommandIntegrationTest.testFileZeroLengthWhenPathNotExist
Error: 1.294 [ERROR]   Run 1: TestCommandIntegrationTest.testFileZeroLengthWhenPathNotExist:133 expected:<1> but was:<3>
Error: 1.294 [ERROR]   Run 2: TestCommandIntegrationTest.testFileZeroLengthWhenPathNotExist:133 expected:<1> but was:<3>
20:24:31.294 [INFO] 
Error: 1.294 [ERROR] alluxio.client.cli.fs.command.TestCommandIntegrationTest.testPathExistWhenPathNotExist
Error: 1.294 [ERROR]   Run 1: TestCommandIntegrationTest.testPathExistWhenPathNotExist:71 expected:<1> but was:<3>
Error: 1.294 [ERROR]   Run 2: TestCommandIntegrationTest.testPathExistWhenPathNotExist:71 expected:<1> but was:<3>
20:24:31.295 [INFO] 
Error: 1.295 [ERROR] alluxio.client.cli.fs.command.TestCommandIntegrationTest.testPathIsDirectoryWhenPathNotExist
Error: 1.295 [ERROR]   Run 1: TestCommandIntegrationTest.testPathIsDirectoryWhenPathNotExist:29 expected:<1> but was:<3>
Error: 1.295 [ERROR]   Run 2: TestCommandIntegrationTest.testPathIsDirectoryWhenPathNotExist:29 expected:<1> but was:<3>
20:24:31.295 [INFO] 
Error: 1.295 [ERROR] alluxio.client.cli.fs.command.TestCommandIntegrationTest.testPathIsFileWhenPathNotExist
Error: 1.296 [ERROR]   Run 1: TestCommandIntegrationTest.testPathIsFileWhenPathNotExist:50 expected:<1> but was:<3>
Error: 1.296 [ERROR]   Run 2: TestCommandIntegrationTest.testPathIsFileWhenPathNotExist:50 expected:<1> but was:<3>
```
@codings-dan Looks like our integration tests failed because of the changes, can you help update the test as well?

**Comment by codings-dan** [comment]
@LuQQiu Thanks for reveiwing my code, I have updated the test file

**Review by gpang — APPROVED** [review]
LGTM

**Comment by apc999** [comment]
alluxio-bot, merge this please

**Pull request merged**